view/frontend/templates/button_product_page.phtml

```diff
@@ -476,5 +476,17 @@ foreach ($block->getAdditionalCheckoutButtonAttributes() as $attrName => $attrVa
             $(document).on('validationcreate', '#product_addtocart_form', function () {
                 $(".bolt-product-checkout-button-disabled").removeClass("bolt-product-checkout-button-disabled")
             });
+
+            // If the form has the novalidate attribute, remove the disabled class. This means that form validation is already initialized,
+            // and we don't need to wait for the validationcreate event to be triggered, as it might have already been triggered before this script loaded.
+            if ($('#product_addtocart_form').is('[novalidate]')) {
+                $(".bolt-product-checkout-button-disabled").removeClass("bolt-product-checkout-button-disabled")
+            }
+
+            // finally, remove the disabled class after 5 seconds if it hasn't been removed yet
```

Comment:
Good question @dumega 
I'm not familiar with M2 validation logic but IMO if we rely on it we shouldn't add any timeout here.
This issue is easily reproductable so let's try to dig down a bit more to find a route cause.

Reply:
I see your point. Nevertheless, we have a merchant website where we can observe this flaky behaviour so let's try to understand the root cause of this issue.
We might consider stopping relying on the `validationcreate` event but this should be analysed as it could have some side effects.